enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+- Move the majority of the current configuration options to a new configuration API - Insights has quite a lot of configuration options
+  and it does not make sense to expose everything in the new configuration API 
+- Removing the original configuration via `support` secret
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
+	// SimpleContentAccessStatus provides basic information about the SCA controller.
+	SimpleContentAccessStatus SimpleContentAccessStatus `json:"simpleContentAccessStatus,omitempty"`
+	// ClusterTransferStatus provides basic information about the ClusterTransfer controller.
+	ClusterTransferStatus ClusterTransferStatus `json:"clusterTransferStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringConfig struct {
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason. Gathering can be forced even
+	// when all the gatherers are disabled.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatheringTime is the last time when Insights gathering finished.
+	LastGatheringTime time.Time `json:"lastGatheringTime,omitempty"`
+	// Reason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	Reason string `json:"reason,omitempty"`
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
```

Comment:
probably worth discussing what happens if someone bumps `forceGatherReason` while a gather is running.  Does that queue up a new gather to start immediately after the current one completes?  Or is that bump ignored?

Is there a timebox on gathers?   If not, touching `forceGatherReason` and having that abort the existing gather and trigger a new one might be a useful way to unwedge a stuck gather.  But there's always "delete the Insights pod, and its replacement won't be stuck", so not a big deal.

Reply:
When there's some gathering in the progress then it will be interrupted. I must emphasize this in the `ForceGatherReason` comment.

Reply:
I updated the comment and the section describing this feature

Reply:
@wking as long as the Reason and ForceGatherReason does not match, there will be always gather queued up to run.